backend/fms_core/template_importer/row_handlers/sample_qc_spark/sample.py

```diff
@@ -0,0 +1,113 @@
+from fms_core.services.process_measurement import create_process_measurement
+from fms_core.services.property_value import create_process_measurement_properties
+from fms_core.services.sample import update_qc_flags
+from fms_core.services.sample_next_step import execute_workflow_action
+from fms_core.models import InstrumentType, Process, Step
+
+INSTRUMENT_PROPERTIES = ['Quantity Instrument']
+QC_PLATFORM = "Quality Control"
+
+SampleDict = TypedDict('SampleDict', {
+    'coordinates': str,
+    'container': Dict[Literal['barcode'], str]
+})
+
+SampleInformationDict = TypedDict('SampleInformationDict', {
+    'concentration': Decimal | None,
+    'quantity_flag': 'False' | 'True' | None
+})
+
+ProcessMeasurementDict = TypedDict('ProcessMeasurementDict', {
+    'process': Process,
+    'execution_date': datetime.date | datetime.datetime | None,
+    'comment': str
+})
+
+ProcessMeasurementPropertiesDict = TypedDict('ProcessMeasurementPropertiesDict', {
+    'Sample Quantity QC Flag': Dict[Literal['value'], str],
+    'Concentration (ng/uL)': Dict[Literal['value'], str],
+})
+
+WorkflowDict = TypedDict('WorkflowDict', {
+    'step_action': str,
+    'step': Step
+})
+
+class SampleQCSparkRowHandler(GenericRowHandler):
+    def __init__(self):
+        super().__init__()
+
+    def process_row_inner(self,
+                          sample: SampleDict,
+                          sample_information: SampleInformationDict,
+                          process_measurement: ProcessMeasurementDict,
+                          process_measurement_properties: ProcessMeasurementPropertiesDict,
+                          workflow: WorkflowDict):
+        sample_obj, self.errors['sample'], self.warnings['sample'] = get_sample_from_container(
+            barcode=sample['container']['barcode'],
+            coordinates=sample['coordinates'],
+        )
+
+        if sample_obj:
+            # Check if sample is not a library or a pool of libraries
+            if sample_obj.is_library:
+                self.errors['source_sample'] = f"Source sample can't be a library or a pool of libraries."
+
+            # Update sample with sample_information
+            if sample_information['concentration'] is None:
+                self.errors['concentration'] = 'Concentration value is required'
+
+            _, self.errors['sample_update'], self.warnings['sample_update'] = \
+                update_sample(sample_to_update=sample_obj,
+                              concentration=sample_information['concentration'])
+
+            # Update the sample's flags with sample information
+            _, self.errors['flags'], self.warnings['flags'] = update_qc_flags(sample=sample_obj,
+                                                                              quantity_flag=sample_information['quantity_flag'],
+                                                                              quality_flag=sample_information['quality_flag'])
```

Comment:
We should change update_qc_flags to take only quality or quantity flags. We can return a warning if both are not set at once. In this case we should not update the quality_flag.